server-side/managers/sharingManager.js

```diff
@@ -0,0 +1,48 @@
+import Users from '../models/user.model';
+import PendingUsers from '../models/pendingUser.model';
+import { createPendingUser } from '../controllers/pendingUser.controler';
+import { createInvitation } from '../controllers/invitation.controler';
+
+export const shareProfile = async (req, res, next) => {
```

Comment:
this is not a controller so you shouldn't get req, res it should be a function that gets normal parameters, and in profiles controller, you should have a function to communicate with the client.